### ISSUE TITLE
Upgrade Node.js to v18

### ISSUE THREAD

**Issue by gzdunek** [gravitational/teleport#21724]
Our builders currently use Node v16 which EOL is [September 11th, 2023](https://nodejs.org/ar/blog/announcements/nodejs16-eol/).
The main problem with the upgrade is that we use CentOS 7 image for Linux pipelines, which Node v18 [does not support](https://nodejs.org/de/blog/announcements/v18-release-announce/#toolchain-and-compiler-upgrades). 
The proposed solution (after moving to GHA) is to use different image for each stage. We can then build tsh on CentOS 7 image and Connect on Ubuntu 22.04. That will be still one pipeline/build, but with multiple different OS-es.
We will have to do something similar for the Web UI.

See the discussion https://gravitational.slack.com/archives/C0DF0TPMY/p1675850357270049

**Comment by ravicious** [comment]
Note on Electron and node-pty. Electron v23 ships with Node v18 already. As [node-pty's docs](https://github.com/microsoft/node-pty#dependencies) say:

> Node.JS 16 or Electron 19 is required to use node-pty. What version of node is supported is currently mostly bound to whatever version Visual Studio Code is using.

The VSCode team is yet to upgrade to Electron v22 (https://github.com/microsoft/vscode/issues/166265). So we have to either wait for them to upgrade to v23 or check if node-pty is indeed incompatible with Node v18. node-pty docs are slightly out of date as node-pty is working just fine with Electron v22.

**Comment by wadells** [comment]
@gzdunek: Did https://github.com/gravitational/teleport/pull/26072 take care of this, or is there still more work to be done for this upgrade?

**Comment by gzdunek** [comment]
No, it didn't take care of it. As I wrote in https://github.com/gravitational/teleport/issues/26856#issuecomment-1563899023 we still have to update the version in the buildboxes.

**Comment by gzdunek** [comment]
Some updates on the Node upgrade:

There is an idea to build Node from source on the systems that don't support it. I experimented with that, here Node is built on centos:7 and it works fine https://drone.platform.teleport.sh/gravitational/teleport/24491/3/1 (it takes around 8 minutes). 

But I talked to @jakule yesterday and we believe that it is not an ideal solution, and a better approach would be to build WebUI and Connect on a newer OS (to be precise, Connect is going to be built on a [different image soon](https://github.com/gravitational/teleport/pull/26905)).

There is a plan to move the rest of our pipeline to GHA in Q2/Q3, so our proposal is to wait, let's say until the end of July, and see if we have GHA for Linux or not. If yes, then we can move building WebUI to run on a different stage then the teleport binary, if not, then we can go to plan B (build from source).

Does it sound good?
